File: src/my_project/easy_problems/from151To200/transpose_matrix.py
from typing import List 

class Solution:
    def transpose(self, matrix: List[List[int]]) -> List[List[int]]:

        flat_matrix = list()
        x = len(matrix)
        y = len(matrix[0])

        for j in range(y):
            for i in range(x):
                flat_matrix.append(matrix[i][j])

        transpose_matrix = list()

        for i in range(y):
            transpose_matrix.append(flat_matrix[i*x:i*x + x])


        return transpose_matrix

solution = Solution()

print(solution.transpose(matrix = [[1,2,3],[4,5,6],[7,8,9]]))

print(solution.transpose(matrix = [[1,2,3],[4,5,6]]))


# function to be reverted 

def functionToBeReverted(num: int)-> int:
    return num *7

File: src/my_project_typescript/easy_problems_typescript/from51To100/transposeMatrix.ts
function transpose(matrix: number[][]): number[][] {

    let flatMatrix: Array<number> = []
    let x: number = matrix.length
    let y: number = matrix[0].length

    for (let j = 0; j < y; j++){
        for (let i = 0; i < x; i++){
            flatMatrix.push(matrix[i][j])
        }
    }

    let transposeMatrix: Array<any> = []

    for (let i = 0; i < y; i++){
        transposeMatrix.push(flatMatrix.slice(i*x, i*x + x))
    }
    return transposeMatrix
};

console.log(transpose([[1,2,3],[4,5,6],[7,8,9]]))

console.log(transpose([[1,2,3],[4,5,6]]))
